component/client/chat/ClientIndependentChatRoom.tsx

```diff
@@ -22,27 +23,24 @@ class ClientIndependentChatRoom extends React.Component<any, IClientIndependentC
     super();
     this.state = {
       currentUrl: Constants.NOOP_URL,
+      userID: Constants.NOOP_ID,
       username: Constants.NOOP_USERNAME
     };
-    storage.get("username")
-      .then((username) => {
-        if (username) {
+    storage.get(Constants.STORAGE_KEY_USERNAME).then((username) => {
+      storage.get(Constants.STORAGE_KEY_ID).then((userID) => {
```

Comment:
Why do you want to do those asynchronous calls in a row?
Get username and get tabs are totally irrelevant. The separate call can make them potentially faster.
I believe if setState() is called often react would do its job to combine some of them, so that it combines several changes together.